raiden/api/rest.py

```diff
@@ -257,6 +258,16 @@ def open(self, partner_address, token_address, settle_timeout, balance=None):
             )
         except (InvalidAddress, InvalidSettleTimeout, SamePeerAddress) as e:
             return make_response(str(e), httplib.CONFLICT)
+        except ValueError as e:
+            if re.match("Token address 0x[0-9a-f]{40} does not contain code", str(e)):
+                return make_response(str(e), httplib.CONFLICT)
+            elif re.match("Manager for token 0x[0-9a-f]{40} does not exist", str(e)):
+                return make_response(str(e), httplib.CONFLICT)
+            raise
+        except Exception as e:
+            if re.match("^Duplicated channel$", str(e)):
```

Comment:
That was @ulope suggestion - e.message is deprecated in P3. See #849 

I agree with comparison vs. regex. - that was just a lazy copypaste by me and I didn't realize doing re.match is not necessary.